src/Instance.ts

```diff
@@ -121,14 +122,14 @@ export function updateTabsterByAttribute(
             case "deloser":
                 if (tabsterOnElement.deloser) {
                     tabsterOnElement.deloser.setProps(
-                        newTabsterProps.deloser as Types.DeloserProps
+                        newTabsterProps as Types.TabsterAttributePropsWith<"deloser">
                     );
                 } else {
                     if (tabster.deloser) {
                         tabsterOnElement.deloser =
                             tabster.deloser.createDeloser(
                                 element,
-                                newTabsterProps.deloser as Types.DeloserProps
+                                newTabsterProps as Types.TabsterAttributePropsWith<"deloser">
```

Comment:
There is no need to pass all props to tabster API, if all we need to use is the Sys API, we can just add that as an argument to the constructor.

This breaks down encapsulation even more and creates really strange utility type to have to maintain

Reply:
I wouldn't call it «breaking the encapsulation». We do separate logic for the particular things, but they still influence each other when nested (like, for example, a groupper state influences the upper mover behaviour). Up to that point we just didn't need to know this influence at the constructor level — we could get it all from the stored props. But now we need it in the constructor, before the props are stored. We set data-tabster for the element, I don't think it's wrong to be able to poke into it in the constructors (as we already do it after the constructors).

Reply:
Another example of the influence closer to this example is Modalizer and Groupper on the same node. The presence of one affects another one.

Reply:
And it seems to be more future-proof to me. I can pass `sys` to every constructor. But then we will add some `supercool` that's required in the constructor and will be forced to pass it down the constructors as well.

Reply:
One more example from the HTML realm. I would compare entire Tabster with `<table>`. Table is a composite entity that consists of a set of intertwined tags and attributes to achieve a layout goal. Tabster is also a set of intertwined things achieve a focus behaviour goal. One part of the table can affect another one and the parts could be correctly or incorrectly composed, like `colspan` and `rowspan` attributes affect the way cells are handled. `<tr>`, `<th>` and `<td>` are different separate parts for different things, but an attribute on one of them might affect each other in different directions. So, the Tabster itself is an encapsulated from the app part, and it consists of the components that are separated from each other code-wise to contain the logic of a particular component and make them maintainable. But they are not isolated components, they serve the singular focus management goal and might be dependent on the composition.